ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/avdtestbase.py

```diff
@@ -43,6 +46,37 @@ def render(self) -> dict:
         """
         return getattr(self, "test_definition", None) or {}
 
+    def get_host_struct_cfg(self, host: str) -> dict:
+        """
+        Retrieves and returns the structured configuration for a specified host.
+
+        The function fetches the structured_config from hostvars. It ensures the returned object is a non-empty
+        dictionary, converting Ansible dictionary-like objects to a standard dictionary if necessary.
+
+        Args:
+            host (str): Hostname to retrieve the structured_config.
+
+        Returns:
+            dict: Structured configuration for the host.
+
+        Raises:
+            AristaAvdError: If host is not in hostvars or if its structured_config is not a dictionary or is empty.
+        """
+        if host not in self.hostvars:
+            raise AristaAvdError(f"Host '{host}' is missing from the hostvars.")
+        struct_cfg = get(self.hostvars, host, separator="..")
+
+        # Check if struct_cfg is a dict or behaves like a dict (e.g. Ansible 'hostvars' object)
+        if not isinstance(struct_cfg, (dict, Mapping)):
+            raise AristaAvdError(f"Host '{host}' structured_config is not a dictionary or dictionary-like object.")
+
+        if not isinstance(struct_cfg, dict):
+            struct_cfg = dict(struct_cfg)
+        if not struct_cfg:
+            raise AristaAvdError(f"Host '{host}' structured_config is empty.")
```

Comment:
The tests that do not need structured_config inputs will still get launched (Hardware, NTP). But the other will be skipped yes:
```
<DC1-BL1> Key 'mlag_configuration' is missing. AvdTestMLAG is skipped.
<DC1-BL1> Key 'router_bgp' is missing. AvdTestBGP is skipped.
[WARNING]: <DC1-BL1> Key 'ethernet_interfaces' is missing. AvdTestLLDPTopology is skipped.
[WARNING]: <DC1-BL1> Key 'ethernet_interfaces' is missing. AvdTestP2PIPReachability is skipped.
[WARNING]: <DC1-BL1> Key 'type' is missing. AvdTestRoutingTable is skipped.
[WARNING]: <DC1-BL1> Key 'management_interfaces' is missing. AvdTestInbandReachability is skipped.
[WARNING]: <DC1-BL1> Key 'type' is missing. AvdTestLoopback0Reachability is skipped.
```
So yeah I will remove this statement since 1. In Ansible, it will never be empty since we have Ansible vars and 2. If it's empty (running outside of Ansible) well we don't want to stop the other tests that do not need anything from the structured_config. A user might want to run the hardware tests only.

Reply:
Fixed in 22836aff3e4102ab512a54b400ec10a4213cff27